### PR TITLE
Retry zfs destroy when busy in rsend tests

### PR DESCRIPTION
**Pull request by dinatale2**
### Description

rsend tests in the test suite frequently create and
destroy datasets. It is possible for zfs destroy to
return an error code indicating the dataset is busy.
Simply use a log_must_busy in these cases to retry
destroying those datasets.

### Motivation and Context

The various `rsend` test failures pointing to a busy dataset which causes a test case to fail.

### How Has This Been Tested?

Testing it with some runs on buildbot to see how this holds up.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux code style requirements.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

File: tests/zfs-tests/tests/functional/rsend/rsend_011_pos.ksh [hunk 1]
```diff
@@ -45,9 +45,6 @@ verify_runnable "both"
 
 function cleanup
 {
-	export __ZFS_POOL_RESTRICT="$POOL $POOL2"
-	log_must zfs unmount -a
-	unset __ZFS_POOL_RESTRICT
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/rsend/rsend_011_pos.ksh]: Why was this removed?

Comment (dinatale2) [inline tests/zfs-tests/tests/functional/rsend/rsend_011_pos.ksh]: Performing that `zfs unmount -a` will unmount `$POOL1`, `$POOL2`, and the datasets within them just to destroy those datasets and remount `$POOL1` and `$POOL2`. This was causing hangs and the unmount and remount doesn't seem necessary.

Comment (behlendorf) [inline tests/zfs-tests/tests/functional/rsend/rsend_011_pos.ksh]: Got it

File: tests/zfs-tests/tests/functional/rsend/rsend.kshlib [hunk 2]
```diff
@@ -296,24 +296,16 @@ function destroy_tree
...
 		typeset ds=${snap%%@*}
 		typeset type=$(get_prop "type" $ds)
 		if [[ $type == "filesystem" ]]; then
 			typeset mntpnt=$(get_prop mountpoint $ds)
-			((ret |= $?))
-
-			if ((ret != 0)); then
-				rm -r $mntpnt/$snap
-				((ret |= $?))
+			if [[ -n $mntpnt ]]; then
```
Comment (tonyhutter) [inline tests/zfs-tests/tests/functional/rsend/rsend.kshlib]: `$mntpnt` should be quoted.  You can also make the test stricter with `if [[ -e "$mntpnt/$snap" ]]`

**Comment by behlendorf** [comment]
I'm going to merge this as-is since to resolve some of the rsend failures we're seeing.  We can followup with another patch to address what's left.
